src/Check/DoctrineMigration.php

```diff
@@ -8,24 +8,51 @@
 
 namespace Laminas\Diagnostics\Check;
 
-use Doctrine\Migrations\Configuration\Configuration;
+use Doctrine\Migrations\DependencyFactory;
+use Doctrine\Migrations\Metadata\AvailableMigration;
+use Doctrine\Migrations\Metadata\ExecutedMigration;
+use InvalidArgumentException;
 use Laminas\Diagnostics\Result\Failure;
 use Laminas\Diagnostics\Result\ResultInterface;
 use Laminas\Diagnostics\Result\Success;
 
 class DoctrineMigration extends AbstractCheck
 {
     /**
-     * @var Configuration
+     * @var array
      */
-    private $migrationConfiguration;
+    private $availableVersions;
 
     /**
-     * @param Configuration $migrationConfiguration
+     * @var array
      */
-    public function __construct(Configuration $migrationConfiguration)
+    private $migratedVersions;
+
+    public function __construct($input)
     {
-        $this->migrationConfiguration = $migrationConfiguration;
+        if ($input instanceof DependencyFactory) {
+            $this->availableVersions = $this->getAvailableVersionsFromDependencyFactory($input);
+            $this->migratedVersions = $this->getMigratedVersionsFromDependencyFactory($input);
+            return;
+        }
+        
+        if ($input instanceof Doctrine\Migrations\Configuration\Configuration
```

Comment:
This is mistake: FQCN should be either prefixed with a backslash or added into use section of the file